debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/EventCache.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.ybexporter;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EventCache {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventQueue.class);
+    private static final String EOF_MARKER = "\\.";
+    String dataDir;
+    private LinkedList<String> eventQueue = new LinkedList<>();
+    private Set<Integer> hashCache = new HashSet<>();
```

Comment:
How about `mostRecentIDFirstList` and `cache` as alternative names.

Note that the `hashCache` and `eventQueue` must record "unique ID" of each event and not the entire event.

Reply:
Done